### PR TITLE
Ase with same friendly name support

### PR DESCRIPTION
**Pull request by hforeste**
+URL changes when having to choose between site that lands in multiple stamps.
eg., sites/jet.com/appanalysis -> select one of the sites -> URL now looks like stamps/{stamp}/sites/{site}/appanalysis
+Make call to new observer admin sites API which takes site name and stamp name

note for some reason I get this error: SideNav 'left' is not available! Did you use md-component-id='left'?
Although when I run AppLens on my local machine the sideNav is working perfectly...

### CODE REVIEW

File: AppLensV2/Controllers/SitesController.cs [hunk 1]
```diff
@@ -48,12 +58,7 @@ public class SitesController : ApiController
                 return ResponseMessage(Request.CreateErrorResponse(hostNameResponse.StatusCode, (string)hostNameResponse.Content));
             }
 
-            if (stampResponse.StatusCode != HttpStatusCode.OK)
-            {
-                return ResponseMessage(Request.CreateErrorResponse(stampResponse.StatusCode, (string)stampResponse.Content));
-            }
-            
-            var resourceGroupResponse =  await SupportObserverClient.GetResourceGroup((string)siteDetailsResponse.Content.First.SiteName);
+            var resourceGroupResponse = await SupportObserverClient.GetResourceGroup((string)siteDetailsResponse.Content.First.SiteName);
```
Comment (stevenernst131) [inline AppLensV2/Controllers/SitesController.cs]: Don't need this anymore

Comment (stevenernst131) [inline AppLensV2/Controllers/SitesController.cs]: As well as the below: siteDetailsResponse.Content.First.ResourceGroupName = resourceGroupResponse.Content;

File: AppLensV2/app/Common/Constants.ts [hunk 2]
```diff
@@ -26,8 +27,12 @@ module SupportCenter {
         private static caseFeedback: string = "/api/cases/{caseId}/feedback";
         private static detectorFeedback: string = "/api/detectors/{detectorName}/feedback";
 
-        public static SiteDetailsPath(siteName: string): string {
-            return UriPaths.siteDetails.replace("{siteName}", siteName);
+        public static SiteDetailsPath(params: IStateParams): string {
+            if (angular.isDefined(params.stamp)) {
```
Comment (ShekharGupta1988) [inline AppLensV2/app/Common/Constants.ts]: Also check for params.stamp !== ''

File: AppLensV2/app/Main/main.html [hunk 3]
```diff
@@ -209,10 +209,11 @@ <h3 style="font-weight:600;font-variant:small-caps;font-size:20px">Please select
             <md-list-item class="md-2-line" ng-repeat="site in multiplesitesctrl.sites">
                 <div class="md-list-item-text" layout="column">
                     <h2 style="margin-bottom: 0 !important">{{::site.name}}</h2>
-                    <h3>{{::site.stampName}}</h3>
+                    <h3>Stamp Name: {{::site.stampName}}</h3>
+                    <h3 ng-hide="{{site.stampName == site.internalStampName}}">Internal Stamp Name: {{::site.internalStampName}}</h3>
```
Comment (ShekharGupta1988) [inline AppLensV2/app/Main/main.html]: use ===

File: AppLensV2/app/MultipleSitesDialog/MultipleSitesDialogCtrl.ts [hunk 4]
```diff
@@ -10,13 +10,14 @@
...
             let locationService = this.$window.location;
             var appLensUrl = locationService.href;
             var oldValue = this.$stateParams.siteName;
-            var newValue = sitename;
-            appLensUrl = appLensUrl.replace(oldValue, newValue);
-            if (this.sites[0].name != sitename) {
+            var newValue = site.name;
+            appLensUrl = locationService.origin + "/stamps/" + site.internalStampName + locationService.pathname.replace(oldValue, newValue) +
+                locationService.search;
+            if (this.sites[0].name != site.name && this.sites[0].internalStampName != site.internalStampName) {
```
Comment (ShekharGupta1988) [inline AppLensV2/app/MultipleSitesDialog/MultipleSitesDialogCtrl.ts]: use !==

File: AppLensV2/app/Main/MainCtrl.ts [hunk 5]
```diff
@@ -251,7 +257,7 @@ module SupportCenter {
 
                 self.properties.push(new NameValuePair("Subscription Id", self.site.subscriptionId));
                 self.properties.push(new NameValuePair("Resource Group", self.site.resourceGroup));
-                self.properties.push(new NameValuePair("Stamp Name", self.site.stampName));
+                self.properties.push(new NameValuePair("Stamp Name", self.site.internalStampName));
```
Comment (stevenernst131) [inline AppLensV2/app/Main/MainCtrl.ts]: should we show both stamp names?

Comment (ShekharGupta1988) [inline AppLensV2/app/Main/MainCtrl.ts]: I agree. maybe show it like waws-prod-...-001(aseprodccs01)

**Pull request merged**